### PR TITLE
Use a bundler for the widget to separate images from code

### LINKED ISSUE #87: Use a bundler for the widget to separate images from code
We should be able to check in image assets and reference them in html using . Then the bundler would automatically inline the image into the output html file.

https://www.figma.com/plugin-docs/libraries-and-bundling/

### PR DESCRIPTION
**Pull request by rylin8**
Use webpack to keep images separate from code so that it is easier to do code reviews and keep images separate. Webpack will inline the images into the output code.

Fixes #87

### CODE REVIEW

**Comment by rylin8** [comment]
If this makes sense I'll do the same for the plugins, since Aubrey has some new designs with lots of images. I'd like to get https://github.com/google/automotive-design-compose/pull/31 in first though because there are some plugin changes there.

**Review by iamralpht — APPROVED** [review]
So nice to see all the assets, and to have a much smaller ui.html file. This will also make it easier to add other JS libraries if we get stuck scaling the plugin up for more complex UIs.

How long does it take to run webpack with this config? Can we get CI to run it to find any missing assets etc?

**Comment by rylin8** [comment]
> So nice to see all the assets, and to have a much smaller ui.html file. This will also make it easier to add other JS libraries if we get stuck scaling the plugin up for more complex UIs.
> 
> How long does it take to run webpack with this config? Can we get CI to run it to find any missing assets etc?

It's fast, I use the "watch" feature in VSCode so it rebuilds whenever it detects any changes and takes about 150ms, so yea we should be able to do that. I get errors in VSCode immediately when I spell an asset name incorrectly in the html file already.

File: support-figma/auto-content-preview-widget/package.json [hunk 1]
```diff
@@ -4,13 +4,23 @@
   "description": "Preview dynamic content in custom layout lists",
   "main": "code.js",
   "scripts": {
-    "build": "tsc -p tsconfig.json"
+    "build": "webpack",
```
[the remark this answers is not in the document]
Comment (rylin8) [reply support-figma/auto-content-preview-widget/package.json:7]: Yes, I'll add that.

File: support-figma/auto-content-preview-widget/manifest.json [hunk 2]
```diff
@@ -2,11 +2,11 @@
   "name": "Auto content preview for DesignCompose",
   "id": "1124100520827492306",
   "api": "1.0.0",
-  "main": "code.js",
-  "ui": "ui.html",
+  "main": "dist/code.js",
```
[the remark this answers is not in the document]
Comment (rylin8) [reply support-figma/auto-content-preview-widget/manifest.json:5]: > How does this change how we upload and install the widget? Do we need to update the instructions in the Readme?

It's basically the same except when you build from VSCode you run `npm watch`. I'll update the readme.

[the remark this answers is not in the document]
Comment (rylin8) [reply support-figma/auto-content-preview-widget/manifest.json:5]: Done

**Comment by timothyfroehlich** [comment]
Could you modify the main readme too? It has instructions for uploading the widget.

**Comment by timothyfroehlich** [comment]
> > Could you modify the main readme too? It has instructions for uploading the widget.
> 
> I don't see any changes needed there. Below are the instructions, and those remain the same.
> 
> ```shell
> npm install
> npm run build
> ```
> 
> Then open the Figma Desktop app, go to **Plugins** -> **Development** -> **Import plugin from Manifest** and select the `manifest.json` file to import.

Oh, ok, I assumed you might need to move files around. Sounds good.

**Comment by timothyfroehlich** [comment]
whoops

**Review by timothyfroehlich — APPROVED** [review]

**Pull request merged**